tests/macros/tquotedo.nim

```diff
@@ -0,0 +1,9 @@
+import macros
+
+macro mac(): untyped =
+  quote do:
+    proc test(): int =
+      (proc(): int = result = 123)()
+
+mac()
+echo test()
```

Comment:
May be use `doAssert` here instead of `echo`?

Reply:
Yes, that would make sense. The original error would prevent this from compiling at all, which is probably why I didn't care to check the actual output of `test`. But it would be good to do it none-the-less.